modules/nextflow/src/main/groovy/nextflow/processor/TaskPollingMonitor.groovy

```diff
@@ -617,6 +619,9 @@ class TaskPollingMonitor implements TaskMonitor {
             if (evict(handler)) { 
                 handler.decProcessForks()
             }
+            //Add trace of the previous execution in the task context for next execution
+            handler.task.config.put(TRACE_PROPERTY_NAME, handler.getTraceRecord())
+            handler.task.config.put(ERROR_PROPERTY_NAME, error.getMessage())
```

Comment:
it would be better to keep this in the `resumeOrDie` method both for consistency and avoid code duplication   

https://github.com/nextflow-io/nextflow/blob/500f49c0911bb33e3fa0f765df4a20da07a8f761/modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L1060-L1063

Reply:
I think no problem for the case of the error. I need to check for the trace record as we need the TaskHandler

Reply:
Done! I have included the traceRecord as optional parameter of the resumeOrDie and inside the function I modify task.config inside the function